theano/tensor/nlinalg.py

```diff
@@ -726,3 +726,62 @@ def norm(x, ord):
             raise ValueError(0)
     elif ndim > 2:
         raise NotImplementedError("We don't support norm witn ndim > 2")
+
+
+class TensorSolve(Op):
+    """
+    Theano utilization of numpy.linalg.tensorsolve
+    Class wrapper for tensorsolve function.
+
+    """
+    _numop = staticmethod(numpy.linalg.tensorsolve)
+    __props__ = ('axes', )
+
+    def __init__(self, axes=None):
+        self.axes = axes
+
+    def make_node(self, a, b):
+        a = as_tensor_variable(a)
+        b = as_tensor_variable(b)
+        out_dtype = theano.scalar.upcast(a.dtype, b.dtype)
```

Comment:
All seem good. But can you add tests of mixed dtype just to make sure this upcast work and that NumPy don't change related to that? They already changed there casting.